DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/devlink/test_devlink_interact.py

```diff
@@ -0,0 +1,377 @@
+
+    dent_dev.applog.info(f'policer rate: {rate_bps // 1000}Kbps, traffic frame size: {frame_size}, '
+                         f'expected trap rate: {exp_rate_pps // 1}pps')
+
+    # 1. Create a bridge entity and set link up on it, enslave all participant ports to the bridge
+    out = await IpLink.add(input_data=[{dent: [
+        {'name': bridge, 'type': 'bridge'}
+    ]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add bridge'
+
+    out = await IpLink.set(input_data=[{dent: [
+        {'device': port, 'operstate': 'up', 'master': bridge}
+        for port in ports
+    ] + [
+        {'device': bridge, 'operstate': 'up'}
+    ]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to enslave ports'
+
+    # 2. Create an ingress qdisc
+    out = await TcQdisc.add(input_data=[{dent: [
+        {'dev': ingress_port, 'direction': 'ingress'}
+    ]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to add qdisc'
+
+    # 3. Generate rule with random selectors
+    selectors = {
+        'want_mac': True, 'want_ip': True, 'want_port': True,
+        'want_tcp': random.randint(0, 1), 'skip_sw': True, 'pref': pref,
+        'action': {'trap': '', 'police': {'rate': f'{rate_bps}bps', 'burst': rate_bps + 1000}}}
+    police_rule = tcutil_generate_rule_with_random_selectors(ingress_port, **selectors)
+
+    drop_rule = deepcopy(police_rule)
+    drop_rule['action'] = 'drop'
+    drop_rule['pref'] *= 2
+
+    # 4. Add two rules with those selectors to the ingress qdisc:
+    out = await TcFilter.add(input_data=[{dent: [
+        police_rule,  # first with police action
+        drop_rule,  # second with action drop
+    ]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to create tc rule'
+
+    # 5. Prepare traffic matching the random selectors
+    dev_groups = tgen_utils_dev_groups_from_config([
+        {'ixp': tg_ports[0], 'ip': '1.1.1.1', 'gw': '2.2.2.2', 'plen': 24},
+        {'ixp': tg_ports[1], 'ip': '2.2.2.2', 'gw': '1.1.1.1', 'plen': 24},
+    ])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    out = await TcFilter.show(input_data=[{dent: [
+        {'dev': ingress_port, 'direction': 'ingress', 'pref': pref, 'options': '-j'}
+    ]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed to get tc rule'
+
+    streams = tcutil_tc_rules_to_tgen_streams(
+        {ingress_port: out[0][dent]['parsed_output']},
+        frame_rate_type='line_rate', frame_rate_pps=100,  # 100%
+        frame_size=frame_size)
+
+    await tgen_utils_setup_streams(tgen_dev, None, streams)
+
+    # 6. Send traffic, don't stop
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+
+    # 7. Verify traffic is trapped by the police rule, and verify the CPU trapped rate for that rule
+    await verify_cpu_traps_rate_code_avg(dent_dev, CPU_STAT_CODE_ACL_CODE_3, exp_rate_pps)
+    await verify_devlink_cpu_traps_rate_avg(dent_dev, 'acl_code_3', exp_rate_pps)
```

Comment:
We can check this in parallel to save some time: 
```
await asyncio.gather(*[verify_cpu_traps_rate_code_avg(dent_dev, CPU_STAT_CODE_ACL_CODE_3, exp_rate_pps, logs=True),
verify_devlink_cpu_traps_rate_avg(dent_dev, 'acl_code_3', exp_rate_pps, logs=True)])
```

Reply:
will do